packages/tokens/src/rwa/compliance/storage.rs

```diff
@@ -0,0 +1,309 @@
+    e.storage().persistent().set(&key, &modules);
+    e.storage().persistent().set(&existence_key, &());
+
+    // Emit event
+    emit_module_added(e, hook, module);
+}
+
+/// Deregisters a compliance module from a specific hook type.
+///
+/// # Arguments
+///
+/// * `e` - Access to the Soroban environment.
+/// * `hook` - The type of hook to unregister the module from.
+/// * `module` - The address of the compliance module contract to remove.
+///
+/// # Events
+///
+/// * topics - `["module_removed", hook: HookType, module: Address]`
+/// * data - `[]`
+///
+/// # Errors
+///
+/// * [`ComplianceError::ModuleNotRegistered`] - When the module is not
+///   currently registered for this hook type.
+///
+/// # Security Warning
+///
+/// **IMPORTANT**: This function bypasses authorization checks and should
+/// only be used:
+/// - During contract initialization/construction
+/// - In admin functions that implement their own authorization logic
+///
+/// Using this function in public-facing methods creates significant
+/// security risks as it could allow unauthorized module removal.
+pub fn remove_module_from(e: &Env, hook: HookType, module: Address) {
+    // Check if module is registered
+    let existence_key = DataKey::ModuleRegistered(hook.clone(), module.clone());
+    if e.storage().persistent().get::<_, ()>(&existence_key).is_none() {
+        panic_with_error!(e, ComplianceError::ModuleNotRegistered);
+    }
+
+    // Get the modules for this hook
+    let key = DataKey::HookModules(hook.clone());
+    let mut modules: Vec<Address> =
+        e.storage().persistent().get(&key).unwrap_or_else(|| Vec::new(e));
+
+    // Remove the module
+    let index = modules.iter().position(|x| x == module).expect("module exists") as u32;
+    modules.remove(index);
+
+    // Update storage
+    e.storage().persistent().set(&key, &modules);
+
+    let existence_key = DataKey::ModuleRegistered(hook.clone(), module.clone());
+    e.storage().persistent().remove(&existence_key);
+
+    // Emit event
+    emit_module_removed(e, hook, module);
+}
+
+// ################## HOOK EXECUTION ##################
+
+/// Executes all modules registered for the Transfer hook.
+///
+/// Called after tokens are successfully transferred from one address to
+/// another. Only modules that have registered for the Transfer hook will be
+/// invoked.
+///
+/// # Arguments
+///
+/// * `e` - Access to the Soroban environment.
+/// * `from` - The address that sent the tokens.
+/// * `to` - The address that received the tokens.
+/// * `amount` - The amount of tokens transferred.
+///
+/// # Cross-Contract Calls
+///
+/// Invokes `on_xfer(from, to, amount)` on each registered module.
+pub fn transferred(e: &Env, from: Address, to: Address, amount: i128) {
+    let modules = get_modules_for_hook(e, HookType::Transfer);
+
+    // Call each registered module
+    for module_address in modules.iter() {
+        let _result: () = e.invoke_contract(
```

Comment:
Why not using a client call instead of the low-level `invoke_contract`? `invoke_contract` also works here, but it seems more suitable for cases where the arguments are unknown at compile time, while here that's not the case.

Reply:
good call, thanks!